clean-modular-code/python-pep-8.md

```diff
@@ -14,237 +14,279 @@ kernelspec:
-
-"""Returns a single sum value of all precipitation. 
-
-This function takes a Pandas DataFrame with time series as the index, 
-and calculates the total sum, aggregated by month. 
-"""
-# Code here 
-
-return the_total_sum
+# This is a class definition
+class MyClass:
+    """A class to process data and calculate statistics."""
+
+    # This is a function 
+    def calculate_average(self, data_list):
+        """Calculate the average of a list of numbers."""
+        total_sum = sum(data_list)
+        count = len(data_list)
+        return total_sum / count
+
+# Example variable names and function call
+data_processor = DataProcessor()
+numbers = [10, 20, 30, 40, 50]
+average_value = data_processor.calculate_average(numbers)
+
+print(f"The average value is: {average_value}")
 ```
 
-## Line Length
-
-PEP 8 guidelines suggest that each line of code (as well as comment lines) should be 79 characters wide or less. This is a common standard used in other languages, including **R**.
+* Line Length: Limit all lines to 79 characters for better readability.
 
 :::{tip}
 Most text editors allow you to set up guides to see how long your code is. You can then use these guides to create line breaks in your code.
 :::
 
+* Comments: Add a space after the `#` sign and capitalize the first letter of a comment:
 
-## Python PEP 8 Rules for White Space
-
-Some of the white space rules have already been discussed above. These including adding a single space after a comment `# Comment here`.
-
-There are also rules associated with spacing throughout your code. These include:
+`# This is a PEP 8 conforming comment`
 
-* **Add blank line before a single line comment (unless it is the first line of a cell in Jupyter Notebook)** Blank lines help to visually break up code. Consider reading this page, if all of the text was mashed together in one long paragraph, it would be more difficult to read. However, when you break the text up into related paragraphs, it becomes a lot easier to read.
+* White Space: Add space between sections of code to improve clarity.
+* Avoid using single-character letters that could be confused with numbers  (ie the letter `l` looks similar to the number one  `1` 
+* Add a blank line before a single-line comment (unless it is the first line of a cell in Jupyter Notebook)
 
 ```python
+a = 1 
 
-# Perform some math
-a = 1+2
-b = 3+4
-c = a+b 
+# Here is a commment
+b = 2 
 
-# Read in and plot some 
-precip_timeseries = pd.readcsv("precip-2019.csv")
-precip_timeseries.plot()
 ```
 
-The code below is more difficult to read as the spacing does not break up the text.
-
-```python
-# Perform some math and do some things 
-a=1+2
-b=3+4
-c=a+b 
-data=pd.readcsv("precip-2019.csv")
-data.plot()
-```
-
-* **Break up sections of code with white space:** As you are writing code, it's always good to consider readability and to break up sections of code accordingly. Breaking up your
+* **Break up sections of code with white space:** Breaking up your
```

Comment:
an idea: all the whitespace stuff could go in a single example like the capitalization stuff

```python
from itertools import counter


# two lines above a function definition
def my_function():
    count = counter()
    values = [next(count) for i in range(10)]

    # one line to break up code into related chunks
    plt.plot(values)


# Two lines before a class definition
class MyClass:
    
    # one line before a method definition
    def my_method(): ...
```